source/extensions/network/dns_resolver/cares/dns_impl.cc

```diff
@@ -123,12 +124,18 @@ void DnsResolverImpl::initializeChannel(ares_options* options, int optmask) {
   }
 }
 
+// Treat responses with `ARES_ENODATA` or `ARES_ENOTFOUND` status as DNS response with no records.
+// @see DnsResolverImpl::PendingResolution::onAresGetAddrInfoCallback for details.
+bool DnsResolverImpl::AddrInfoPendingResolution::isResponseWithNoRecords(int status) {
+  return status == ARES_ENODATA || status == ARES_ENOTFOUND;
```

Comment:
I feel it should be better to have the run time flag check within this function, like:

return (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.cares_accept_nodata") &&
            ( status == ARES_ENODATA || status == ARES_ENOTFOUND));

That will make the code outside more readable, like in onAresGetAddrInfoCallback().
This way, you can also avoid adding  accept_nodata_.

Reply:
Ack, I think I saw that there were also some question on getting rid of accept_nodata_, I wanted to keep it because:

1. To save a call to `Runtime:: runtimeFeatureEnabled`
2. Make sure feature flag won't change while we process the request, I don't think anything would be broken even if this happens but just in case things/code become more complex in the future.